### ISSUE TITLE
Allow a | to be used as a Valid Id in Argument.cs

### ISSUE THREAD

**Issue by kj4bix** [Azure/azure-mobile-apps#672]
I'm moving an application from the azure mobile client sdk. The existing client SDK allows a | to be used in an Id column. 

The new Microsoft.Datasync.Client doesn't allow the | character to be used. This regex is in Microsoft.Datasync.Client/Utils/Arguments.cs, named validIdRegex 

private static readonly Regex validIdRegex = new("^[a-zA-Z0-9][a-zA-Z0-9-_.]{0,126}$");

To me it seems any string that is acceptable in the supported databases should be allowed.

**Comment by adrianhall** [comment]
There are specific reasons why "any string" is not allowed, most notably to do with JSON encoding of strings.  The ID definitely has to be UTF-8, but we are too restrictive.

TODO:
[ ]: Update Arguments.cs to allow other symbols.
[ ]: Consider adding an option in CreateAsync to bypass the ID check - what's the effect?
[ ]: Update Arguments.Test.cs to check for validity.

**Comment by kj4bix** [comment]
I changed the RegEx to include the | specifically - rebuilt and replaced the datasync.client dll and it works. 
However, I like the idea of adding an option to bypass the check or maybe just a more inclusive check. It worked in the older SDK. If it doesn't serialize to the JSON string, they would get an error from the serialization itself (basically something that isn't possible to encode). 

Here is what I changed it to on my machine: 
"^[a-zA-Z0-9][a-zA-Z0-9-_.|]{0,126}$"

**Comment by adrianhall** [comment]
I think I can make it the following:

``` csharp
private static readonly Regex validIdRegex = new("^[a-zA-Z0-9][\x21-\x7F]{0,126}$");
```

This ensures the first character is suitable for anything and anything UTF-8 printable except for space can be used.  This is the ID format most commonly used.  I'll also include an override in the options to allow an alternate regular expression to be used.

**Comment by kj4bix** [comment]
Great, I'll try it on my copy.

**Comment by adrianhall** [comment]
Released in NuGet: 5.1.4 - wait a couple of hours for the release to show up.
